Documentation/contributing/docs/docsframework.rst

```diff
@@ -0,0 +1,382 @@
+  - Generates ``Documentation/helm-values.rst``
+
+- ``update-codeowners``
+
+  - ``./update-codeowners.sh``
+  - Synchronizes teams description from ``CODEOWNERS``
+  - Generates ``Documentation/codeowners.rst``
+
+Other auto-generated contents include:
+
+- OpenAPI reference
+
+  - YAML generated from the ``Makefile`` at the root of the repository
+  - Relies on the contents of ``api``, linked as ``Documentation/_api``
+  - Processed and included via a dedicated add-on, from
+    ``Documentation/api.rst``: ``.. openapi:: ../api/v1/openapi.yaml``
+
+- gRPC API reference
+
+  - Markdown generated from the main ``Makefile`` at the root of the repository
+  - Relies on the contents of ``api``, linked as ``Documentation/_api``
+  - Included from ``Documentation/grpcapi.rst``
+
+Build system
+============
+
+Makefile targets
+----------------
+
+Here are the main ``Makefile`` targets related to documentation, to run from
+the root of the Cilium repository, as well as some indications on what they
+call:
+
+- ``make`` -> ``all: ... postcheck`` -> ``make -C Documentation check``:
+  Build Cilium and validate the documentation via the ``postcheck`` target
+- ``make -C Documentation html``:
+  Render the documentation as HTML
+- ``make test-docs`` -> ``make -C Documentation html``:
+  Render the documentation as HTML
+- ``make -C Documentation live-preview``:
+  Build the documentation and start a server for local preview
+- ``make render-docs`` -> ``make -C Documentation live-preview``:
+  Build the documentation and start a server for local preview
+
+Generating documentation
+------------------------
+
+- The ``Makefile`` builds the documentation using the ``docs-builder`` Docker
+  image.
+
+- The build includes running ``check-build.sh``. This script:
+
+  a. Runs the linter (``rstcheck``) by default
+  b. Runs the spell checker
+  c. Builds the HTML version of the documentation
+  d. Exits with an error if any unexpected warning or error is found
+
+Tweaks and tools
+================
+
+See also file ``Documentation/conf.py``.
+
+Spell checker
+-------------
+
+We rely on Sphinx's `spell-checker module`_ (considered a "`builder`_" in
+Sphinx).
+
+The spell checker users a list of known exceptions, contained in
+``Documentation/spelling_wordlist.txt``. Words in that list that are written
+with lowercase exclusively, or uppercase exclusively, are case-insensitive
+exceptions for spell-checking. Words with mixed case are case-sensitive. This
+file should remain sorted alphabetically.
+
+To add new entries to the list, run ``Documentation/update-spelling_wordlist.sh``.
+
+To clean-up obsolete entries, make sure the spell checker reports no issue on
+the current version of the documentation, then remove all entries from the
+file, run the spell checker and re-add all reported exceptions.
+
+We use a custom filter for the spell checker, for spelling ``WireGuard``
+correctly as ``WireGuard``, or ``wireguard`` in some contexts, but never as
```

Comment:
Clarity nit
```suggestion
We use a custom filter for the spell checker, for example: spelling ``WireGuard``
correctly as ``WireGuard``, or ``wireguard`` in some contexts, but never as
```

Reply:
This is not an example, this is the exhaustive behaviour of all this one custom filter we have :smile:

Reply:
Wow! I withdraw the change. 😅